jaxrs2/src/main/java/feign/jaxrs2/JAXRS2Contract.java

```diff
@@ -16,23 +16,20 @@
 import javax.ws.rs.container.Suspended;
 import javax.ws.rs.core.Context;
 import feign.jaxrs.JAXRSContract;
-import java.lang.annotation.Annotation;
 
 /**
  * Please refer to the <a href="https://github.com/Netflix/feign/tree/master/feign-jaxrs2">Feign
  * JAX-RS 2 README</a>.
  */
 public final class JAXRS2Contract extends JAXRSContract {
-  @Override
-  protected boolean isUnsupportedHttpParameterAnnotation(Annotation parameterAnnotation) {
-    Class<? extends Annotation> annotationType = parameterAnnotation.annotationType();
 
-    // masc20180327. parameter with unsupported jax-rs annotations should not be passed as body
-    // params.
+  public JAXRS2Contract() {
+    // parameter with unsupported jax-rs annotations should not be passed as body params.
     // this will prevent interfaces from becoming unusable entirely due to single (unsupported)
     // endpoints.
     // https://github.com/OpenFeign/feign/issues/669
-    return (annotationType == Suspended.class ||
-        annotationType == Context.class);
+    super.registerParameterAnnotation(Suspended.class, (ann, data, i) -> true);
```

Comment:
If they are not supported, then do we even need to register them?

Reply:
You need to let the contract parser know that this parameters should have an special handling. In this case, should be ignored.

If nothing is done feign will assume that is the body, failing of there is more than one.

Reply:
Oh, right, I remember now.  Thanks.